xdsl/dialects/experimental/dmp.py

```diff
@@ -128,17 +125,17 @@ def elem_count(self) -> int:
         return prod(self.size)
 
     @property
-    def dim(self) -> int:
+    def rank(self) -> int:
         return len(self.size)
```

Comment:
Why should a rank return len(self.size?)

Reply:
Good catch! I guess *rank* is an overloaded term here. It should probably be `dims` to be less ambigous. I'll add a docstring too